packages/commonwealth/server/routes/chat/getChatChannel.ts

```diff
@@ -14,17 +21,65 @@ export const Errors = {
  * @param res
  * @param next
  */
-export default async (models: DB, req: Request, res: Response, next: NextFunction) => {
-	// check channel id
-	if (!req.query.channel_id) {
-		return next(new AppError(Errors.NoChannelId))
-	}
+export default async (
+  models: DB,
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  // check channel id
+  if (!req.query.channel_id) {
+    return next(new AppError(Errors.NoChannelId));
+  }
+  //   check chain id
+  if (!req.query.chain_id) {
+    return next(new AppError(Errors.NoChainId));
+  }
+  //   If not logged in check if the channel is public
+  if (!req.user?.id) {
```

Comment:
We need to be checking based on whether or not `getActiveAddress` returns something or nothing -- i.e. if there IS an activeAddress, we use `isAddressPermitted`. Otherwise, we use `isAnyonePermitted`.

This entire pattern should be abstracted away into a single check.

Reply:
put this in a helper in the server utils!